keras_cv/models/backbones/vit/vit_backbone.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""ViT (Vision Transformer) models for Keras.
+Reference:
+  - [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929v2)
+    (ICLR 2021)
+  - [How to train your ViT? Data, Augmentation, and Regularization in Vision Transformers](https://arxiv.org/abs/2106.10270)
+    (CoRR 2021)
+"""  # noqa: E501
+
+import copy
+
+from tensorflow import keras
+from tensorflow.keras import layers
+
+from keras_cv.layers import TransformerEncoder
+from keras_cv.layers.vit_layers import PatchingAndEmbedding
+from keras_cv.models import utils
+from keras_cv.models.backbones.backbone import Backbone
+from keras_cv.models.backbones.vit.vit_backbone_presets import backbone_presets
+from keras_cv.models.backbones.vit.vit_backbone_presets import (
+    backbone_presets_with_weights,
+)
+from keras_cv.utils.python_utils import classproperty
+
+
+@keras.utils.register_keras_serializable(package="keras_cv.models")
+class ViTBackbone(Backbone):
+    """Instantiates the ViT architecture.
+
+    Reference:
+        - [An Image is Worth 16x16 Words: Transformers for Image Recognition at Scale](https://arxiv.org/abs/2010.11929v2)
+        (ICLR 2021)
+    This function returns a Keras {name} model.
+
+    The naming convention of ViT models follows: ViTSize_Patch-size
+        (i.e. ViTS16).
+    The following sizes were released in the original paper:
+        - S (Small)
+        - B (Base)
+        - L (Large)
+    But subsequent work from the same authors introduced:
+        - Ti (Tiny)
+        - H (Huge)
+
+    The parameter configurations for all of these sizes, at patch sizes 16 and
+    32 are made available, following the naming convention laid out above.
+
+    For transfer learning use cases, make sure to read the
+    [guide to transfer learning & fine-tuning](https://keras.io/guides/transfer_learning/).
+
+    Args:
+        mlp_dim: the dimensionality of the hidden Dense layer in the transformer
```

Comment:
nit: please keep these ordered the same as they appear in the constructor's parameter list

Reply:
Understood, done!